working/0546-patterns/records-feature-specification.md

```diff
@@ -322,38 +322,39 @@ through calls to `identical()`, then optimizing away the creation of these
 objects is harder.
 
 Semantically, we do not want records to have unique identities distinct from
-their contents. A record *is* its contents in the same way that every value 3
-in a program is the "same" 3 whether it came from the number literal `3` or the
-result of `1 + 2`.
+their contents. A record *is* its contents in the same way that every value 3 in
+a program is the "same" 3 whether it came from the number literal `3` or the
+result of `1 + 2`. This is why `==` for records is defined in terms of their
+shape and fields. Two records with the same shape and equal fields are equal
+values.
 
-This is why `==` for records is defined in terms of their shape and fields. Two
-records with the same shape and fields are equivalent. Identity follows similar
-rules. Calling `identical()` with a record argument returns:
+At the same time, we want `identical()` to be fast because one of its primary
+uses is as a fast-path check for equality. An `identical()` that is obliged to
+iterate over the record's fields (transitively in the case where some fields
+are themselves records) might nullify the benefits of using `identical()` as a
+fast-path check before calling `==`.
+
+To balance those opposing goals, `identical()` on records is defined to only
+offer loose guarantees. Calling `identical()` with a record argument returns:
 
 *   `false`, if the other argument is not a record.
 *   `false`, if the records do not have the same shape. *Since named field
     order is not part of a record's shape, this implies that named field order
     does not affect identity either. `(a: 1, b: 2)` and `(b: 2, a: 1)` are
     identical.*
```

Comment:
Not sure if it's necessary to do for this feature spec, but I agree that separating the two concepts might be a cleaner way to specify this, perhaps in the formal spec.  Though I would avoid confusion by choosing a new term: e.g. define a notion of "same object", and define the identical function in terms of "same object".  cc @eernstg

Reply:
I like the idea of defining "same object" as the optimal identical behavior and `identical()` as a fast-path version of that that may report false negatives. But I don't want to define that here since it should take into account objects of other types.

Reworded the non-normative text to not say "are identical".